service/node.go

```diff
@@ -2660,10 +2660,25 @@ func (s *service) NodeExpandVolume(
 
 	// Get the pmax volume name so that it can be searched in the system
 	// to find mount information
-	replace := CSIPrefix + "-" + s.getClusterPrefix() + "-"
-	volName := strings.Replace(vol.VolumeIdentifier, replace, "", 1)
-	// remove the namespace from the volName as the mount paths will not have it
-	volName = strings.Join(strings.Split(volName, "-")[:2], "-")
+	// Examples of possible volumeIdentifiers: tn1-csi-ABC-csm-3f7da6bf8d-test, csi-ABC-csm-3f7da6bf8d-test
+	parts := strings.Split(vol.VolumeIdentifier, "-")
+
+	var volName string
+	clusterPrefix := s.getClusterPrefix()
+	// remove the tenant volume prefix (csm-authorization) and namespace from the volName as the mount paths will not have it
+	for i, p := range parts {
```

Comment:
Can lines 2669 to 2681 be added to a separate function so that it is easier to unit test? It would also help to test other variations of volumeIdentifiers such as if the Auth tenant prefix is also "csi" (e.g. csi-csi-csm-3f7da6bf8d-test) or where a hyphen is used in the Tenant prefix (e.g. my-tenant-csi-csm-3f7da6bf8d-test).